### DISCUSSION TITLE
User defined seed & migrations

### DISCUSSION THREAD

**Discussion by maartenvn** [directus/directus#5712] [Feature Requests]
There is currently no easy way to handle migration between a dev/staging area to production, without writing some external migration wrapper around Directus. The import/export endpoint could be a solution, but that would still require manual exporting on one instance and than manually importing on the other.

What if we approached this differently, just like the current seeding & migrating of the Directus specific tables? You would be able to provide seeds for the different database tables and migrations to migrate between the different versions of the application, straight from a VCS (I am thinking YAML files, just like the core). Directus would check at startup if there is a new, user defined, version available and would automatically run the required migrations.

This way you could have a single-source of truth on a platform like GitHub, without requiring any external tools or wrappers.

This feature would not be mandatory, but  rather an optional feature. In this case the developer of the Directus application would not manage the tables, permissions, ... through the admin UI, but would rather work directly from the seed/migration files.  It is a different flow for a different use-case, where the application is not necessarily managed by a fixed group of people (eg: open-source project that is using Directus)

**Comment by rijkvanzanten** [comment]
Duplicate of https://github.com/directus/directus/discussions/3891 🙂

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
